### DISCUSSION TITLE
Error: No Serverless Pages Built

### DISCUSSION THREAD

**Discussion by nugthan** [vercel/community#1360] [Help] [answered]
### Summary

Hey everyone,

I've been using Vercel and Nextjs for quite some time now, however I have never experienced this error before.

![image](https://user-images.githubusercontent.com/37552537/214803560-15bf06bb-df75-4af7-9d5c-ffb93338ad39.png)

The help article linked beneath is outdated and none of the solutions work for me. I have already tried hours of problem solving into what the issue is before making this.

When I build locally, it works fine. If I use the Vercel CLI to build and manually upload to Vercel, its fine, however having Vercel do what it normally does and how I've been using it for years, doesn't work.

I have tried the solutions listed in the only other post on here about it, and I can't find any packages that are not listed in my dependencies file that should be there. I may be missing some, but its something I've already looked into.

Even in my MR version without any serverless functions in it fails as well, which baffles me.

For any Vercel staff: https://vercel.com/nugthan/fdmd-error

Thanks in advance.

### Example

https://github.com/nugthan/fdmd-error

**Comment by swarnava** [comment]
Looks like `pages` folder is missing which is throwing the error. How are you generating the project? using `npx create-next-app@latest`?

**Reply by nugthan** [reply]
> Looks like `pages` folder is missing which is throwing the error. How are you generating the project? using `npx create-next-app@latest`?

Its in the /src folder, which is a valid way of doing things in the nextjs docs. I've tried moving everything into the root directory as well with the same error occurring.

**Reply by swarnava** [reply]
Are you able to run `npm run build` locally?

**Reply by nugthan** [reply]
> Are you able to run `npm run build` locally?

Yes, everything runs perfectly fine locally

"When I build locally, it works fine. If I use the Vercel CLI to build and manually upload to Vercel, its fine, however having Vercel do what it normally does and how I've been using it for years, doesn't work."

**Reply by swarnava** [reply]
When I run the project locally, I get stuck at `info  - Creating an optimized production build` steps:

<img width="814" alt="image" src="https://user-images.githubusercontent.com/1162991/214897748-db3f969a-10a7-457e-89cf-adfc22511cf0.png">

**Reply by nugthan** [reply]
Well considering it's working absolutely fine for me, what would you recommend I do based on that information?

**Reply by leerob** [reply]
A few things that look off that might be related: your `next.config.js` file is empty, you can just remote it entirely. Second, you're using Sass but it doesn't appear installed as a dependency: https://nextjs.org/docs/basic-features/built-in-css-support#sass-support

**Reply by nugthan** [reply]
> A few things that look off that might be related: your `next.config.js` file is empty, you can just remote it entirely. Second, you're using Sass but it doesn't appear installed as a dependency: https://nextjs.org/docs/basic-features/built-in-css-support#sass-support

Hey, thanks for the reply. Done these changes on my main build, still getting the same error, just without the config warning.
![image](https://user-images.githubusercontent.com/37552537/214937924-e6d9e6a8-ae41-4370-b575-b471903ede35.png)

I will do these changes on the MR version as well.

**Reply by swarnava** [reply]
Nevermind the hang issue seems related to https://github.com/vercel/next.js/issues/44856

**Comment by nugthan** [comment, marked as answer]
Right, I'm not _really_ sure what I changed but I managed to get it to work. It may have had something to with sass, may have been tailwind, however its working now so that's good.

Thanks for the help everyone
